modules/Bio/EnsEMBL/IO/Translator/BulkFetcherFeature.pm

```diff
@@ -0,0 +1,515 @@
+    unless $args{ontology_adaptor}->isa('Bio::EnsEMBL::DBSQL::OntologyTermAdaptor');
+
+  croak "Bio::EnsEMBL::IO::Translator::BulkFetcherFeature requires a meta adaptor"
+    unless $args{meta_adaptor} and $args{meta_adaptor}->isa('Bio::EnsEMBL::DBSQL::MetaContainer');
+  
+  $args{ontology_cache} = {};
+  $args{mapping} = $xref_mapping;
+  $args{biotype_mapper} = $biotype_mapper;
+  
+  my $self = $class->SUPER::new(\%args);
+  $self->version($args{version});
+  $self->production_name($self->meta_adaptor->list_value_by_key('species.production_name')->[0]);
+  
+  # once we have the instance, add our customized callbacks to the translator
+  $self->add_callbacks(\%field_callbacks);
+
+  return $self;
+}
+
+sub version {
+  my ($self, $version) = @_;
+  
+  $self->{version} = $version if $version;
+  return $self->{version};
+}
+
+sub production_name {
+  my ($self, $prod_name) = @_;
+
+  $self->{production_name} = $prod_name if $prod_name;
+  return $self->{production_name};
+}
+
+sub ontology_cache {
+  my $self = shift;
+  return $self->{ontology_cache};
+}
+
+sub ontology_adaptor {
+  my $self = shift;
+  return $self->{ontology_adaptor};
+}
+
+sub meta_adaptor {
+  my $self = shift;
+  return $self->{meta_adaptor};
+}
+
+sub ensembl_mapper {
+  my $self = shift;
+  return $self->{mapping};
+}
+
+sub biotype_mapper {
+  my $self = shift;
+  return $self->{biotype_mapper};
+}
+
+sub id_org_short {
+  my ($self, $db) = @_;
+  defined $db or croak "Undefined db";
+  
+  return $self->ensembl_mapper->identifier_org_short($db);
+}
+
+sub lod_uri {
+  my ($self, $db) = @_;
+  defined $db or croak "Undefined db";
+
+  return $self->ensembl_mapper->LOD_uri($db);
+}
+
+=head2 type
+
+    Description: Wrapper around RDF feature type field
+    Returntype : String
+
+=cut
+
+sub type {
+  my ($self, $object) = @_;
+
+  exists $object->{ensembl_object_type} and defined $object->{ensembl_object_type}
+    or croak "Undefined type for feature ", $self->id($object);
+
+  return $object->{ensembl_object_type};
+}
+
+=head2 id
+
+    Description: Wrapper around RDF feature id field
+    Returntype : String
+
+=cut
+
+sub id {
+  my ($self, $object) = @_;
+  exists $object->{id} and defined $object->{id}
+    or croak "Feature does not have id";
+  
+  return $object->{id};
+}
+
+=head2 name
+
+    Description: Wrapper around RDF feature name field
+    Returntype : String
+
+=cut
+
+sub name {
+  my ($self, $object) = @_;
+  
+  return $object->{name};
+}
+
+=head2 description
+
+    Description: Wrapper around RDF feature description field
+    Returntype : String
+
+=cut
+
+sub description {
+  my ($self, $object) = @_;
+  
+  return $object->{description};
+}
+
+
+=head2 seq_region_name
+
+=cut
+
+sub seq_region_name {
+  my ($self, $object) = @_;
+  
+  return $object->{seq_region_name};
+}
+
+=head2 coord_system_name
+
+=cut
+
+sub coord_system_name {
+  my ($self, $object) = @_;
+  
+  return $object->{coord_system}{name};
```

Comment:
You're right, care should be taken at one point or another. Nevertheless, in this particular case, I think we're safe to favor speed as this emanates from a direct contract stipulated with Dan S. He's already promptly informed of an intended change to the underlying data structure.